File: JiHwan/1759.py
import sys

def DFS(depth,start):
    vo = 0
    co = 0
    if depth == n :
        for  i in range(n):
            if vowel[i] in answer:
                vo +=1
            else:
                co +=1
        
            if vo >=1 and co >= 2:
                print(''.join(answer))

    for i in range(0,c):
        answer.append(word[i])
        DFS(depth+1,start+1)
        answer.pop()


    



n ,c = map(int, sys.stdin.readline().split())
word = (list(sys.stdin.readline().split()))
word.sort()
vowel = ['a','e','i','o','u']
answer = []

DFS(0,0)

File: SeongYeon/ts/baekjoon/1759.ts
// input
const input = require('fs')
  .readFileSync('test/input.txt')
  .toString()
  .trim()
  .split('\n');

// variable
let result: string = '';
const [n, m]: number[] = input[0].split(' ').map((a: string) => +a);
const wordList: string[] = input[1].split(' ').sort();
const vowels: string[] = ['a', 'e', 'i', 'o', 'u'];
const tempList: string[] = [];

// solution
const dfs = (depth: number, start: number, numOfVowel: number) => {
  if (depth === n) {
    if (numOfVowel > 0 && numOfVowel <= n! - 2)
      result += tempList.join('') + '\n';
    return;
  }
  for (let i = start; i < m!; i++) {
    tempList[depth] = wordList[i]!;
    if (vowels.includes(wordList[i]!)) dfs(depth + 1, i + 1, numOfVowel + 1);
    else dfs(depth + 1, i + 1, numOfVowel);
  }
};

dfs(0, 0, 0);

// output
console.log(result);
